gonzo/clouds/compute.py

```diff
@@ -146,8 +147,47 @@ def create_instance(self, image_name, name, owner, user_data=None,
         )
         self.compute_session.wait_until_running([instance])
         new_instance = self.get_instance_by_uuid(instance.uuid)
+
+        if volume_size is not None:
+            self.create_and_attach_volume(new_instance, volume_size)
+
         return new_instance
 
+    def create_and_attach_volume(self, instance, vol_size, vol_type=None,
+                                 vol_name=None):
+        created_volume = self.create_volume(
+            instance, instance.name,
+            vol_size, vol_type)
+
+        self.wait_until_volume_available(created_volume)
+
+        self.compute_session.attach_volume(
+            node=instance,
+            volume=created_volume,
+            device='/dev/xvdf'
+        )
+
+    def wait_until_volume_available(self, volume):
+        while not self.volume_is_created(volume):
+            print "Volume Being Created"
+        return True
+
+    def volume_is_created(self, volume):
+        for vol in self.compute_session.list_volumes():
+            if vol.id == volume.id:
+                if vol.extra['state'] == 'available':
+                    return True
```

Comment:
you probably want to distinguish "still pending" (return false) from "unknown volume" (raise exception?)